pkg/ddevapp/mutagen.go

```diff
@@ -433,12 +434,14 @@ func (app *DdevApp) GetMutagenSyncID() (id string, err error) {
 // MutagenSyncFlush performs a mutagen sync flush, waits for result, and checks for errors
 func (app *DdevApp) MutagenSyncFlush() error {
 	status, _ := app.SiteStatus()
-	if status == SiteRunning && app.IsMutagenEnabled() {
+	// TODO: Avoid this failure by looking at all statuses?
+	if (status == SiteRunning || status == SiteUnhealthy) && app.IsMutagenEnabled() {
```

Comment:
This was the fundamental change. In this context (`ddev mutagen reset && ddev start -y`) the web container is not yet healthy, because mutagen hasn't finished syncing. And sadly we then skipped out of here.

Discussion needed on how to make this more robust. I think probably we should just fail if site is not in expected situation. OTOH this condition was put here for a. reason I'm sure.